### PR TITLE
Implement Basic QFileSystemWatcher

### LINKED ISSUE #1858: Live Viewer: Explore implementation of QFileSystemWatcher
Explore Implementation of QFileSystemWatcher based on findings found by  #1844.

### PR DESCRIPTION
**Pull request by JackEAllen**
### Issue

Closes #1858 

### Description

Implement QFileSystemWatcher to add the latest image from a directory to the live viewer window and removes images form the view as they are removed from the directory. 

### Acceptance Criteria 

* As new files arrive in a directory being watched, images added to the live viewer.
* If files are removed from live viewer

### Documentation
See Next Release Notes

### CODE REVIEW

File: mantidimaging/gui/windows/live_viewer/presenter.py [hunk 1]
```diff
@@ -30,3 +32,31 @@ def __init__(self, view: LiveViewerWindowView, main_window: MainWindowView):
...
+        self.model.path = path
+
+    def update_image(self, image_path: str):
+        """Update the image in the view."""
+        if not Path(image_path).exists():
+            return
+        try:
+            with tifffile.TiffFile(image_path) as tif:
+                image_data = tif.asarray()
+                self.view.show_most_recent_image(image_data)
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/live_viewer/presenter.py:47]: Could use `self.view.show_image(image_data)` and remove `view.show_most_recent_image` wrapper.

File: mantidimaging/gui/windows/live_viewer/live_view_widget.py [hunk 2]
```diff
@@ -29,3 +29,9 @@ def show_image(self, image: np.ndarray) -> None:
             @param image: The image to show
             """
         self.image.setImage(image)
+
+    def handle_deleted(self):
+        print(f"Closing image {self.image}")
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/live_viewer/live_view_widget.py:34]: Can be removed

File: mantidimaging/gui/windows/live_viewer/presenter.py [hunk 3]
```diff
@@ -30,3 +32,31 @@ def __init__(self, view: LiveViewerWindowView, main_window: MainWindowView):
...
+        """Display the last modified image"""
+        if not Path(image_path).exists():
+            return
+        try:
+            with tifffile.TiffFile(image_path) as tif:
+                image_data = tif.asarray()
+                self.view.show_last_modified_image(image_data)
+        except IOError as error:
+            logger.error("Error reading image: %s", error)
+            return
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/live_viewer/presenter.py:62]: I don't think this is used. Just a duplicate of `update_image()`

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
Looks good, a few code tweaks

File: mantidimaging/gui/windows/live_viewer/presenter.py (hunk 1 above)
Comment (JackEAllen) [reply mantidimaging/gui/windows/live_viewer/presenter.py]: Thanks for noticing this, yes, that's correct

File: mantidimaging/gui/windows/live_viewer/presenter.py (hunk 3 above)
Comment (JackEAllen) [reply mantidimaging/gui/windows/live_viewer/presenter.py]: Nice catch! this was left over from previous refactoring. have removed.

File: mantidimaging/gui/windows/live_viewer/view.py [hunk 4]
```diff
@@ -33,15 +32,26 @@ def __init__(self, main_window: 'MainWindowView', live_dir_path: Path) -> None:
...
+    def watch_directory(self):
+        """Show the most recent image arrived in the selected directory"""
+        self.presenter.set_dataset_path(self.path)
+
+    def show_most_recent_image(self, image_path: np.ndarray) -> None:
+        """
+        Show the most recent image arrived in the selected directory
+        @param image_file_path: The path of the image to show
+        """
+        self.show_image(image_path)
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/live_viewer/view.py:53]: I think this is not used any more now

Comment (JackEAllen) [reply mantidimaging/gui/windows/live_viewer/view.py]: Thanks for catching this also, same refactoring missed as previous

**Review by samtygier-stfc — APPROVED** [review]
Working great. Thanks.

**Comment by JackEAllen** [comment]
> Working great. Thanks.

Thanks for the review! 👍

**Pull request merged**